packages/openid4vc/src/openid4vc-holder/OpenId4VciHolderService.ts

```diff
@@ -3,82 +3,65 @@ import type {
   OpenId4VciAuthCodeFlowOptions,
```

Comment:
Not really super concrete feedback but this file is getting very large and has some complex to read error handling. It might be good in the future to just create single use functions for the validation, i.e. `getDPoPSigningAlgorithm`, which handles the error as well. Reading the main functionality is a lot easier then.

Reply:
Yes agreed -- it's a mostly from the pattern we had that all logic is in the service, but it's getting a bit out of hand. Do you want me to tackle as part of this PR?

Reply:
Can be a separate PR :)